madlibs.js

```diff
@@ -22,21 +22,110 @@
  *  { word: "store", pos: "noun" }
  *  { word: "," }
  *  ....
- * 
+ *
  * There are multiple ways to do this, but you may want to use regular expressions.
  * Please go through this lesson: https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/regular-expressions/
  */
+
 function parseStory(rawStory) {
-  // Your code here.
-  return {}; // This line is currently wrong :)
+  let arr = rawStory.split(" ");
+  let newArr = arr.map((ele) => {
+    if (ele.includes("[")) {
+      // if it contains [
+      let pos;
+      let index = ele.indexOf("[");
+      let word = ele.slice(0, index); // take the word
+      let regex = ele.match(/\[(.*?)\]/)[1]; //take the character inside the square //Dot . =>match any character except newlines     Quantifier * => Match 0 or more of the preceding token.  Lazy ? =>Makes the preceding quantifier lazy, causing it to match as few characters as possible.
+      if (regex === "v") {
+        pos = "verb";
+      } else if (regex === "n") {
+        pos = "noun";
+      } else if (regex === "o") {
+        pos = "object";
+      } else if (regex === "a") {
+        pos = "adjective";
+      }
+      return { word: word, pos: pos };
+    } else {
+      return { word: ele };
+    }
+  });
+  return newArr; // new array
```

Comment:
```suggestion
  return arr.map((ele) => {
    if (ele.includes("[")) {
      // if it contains [
      let pos;
      let index = ele.indexOf("[");
      let word = ele.slice(0, index); // take the word
      let regex = ele.match(/\[(.*?)\]/)[1]; //take the character inside the square //Dot . =>match any character except newlines     Quantifier * => Match 0 or more of the preceding token.  Lazy ? =>Makes the preceding quantifier lazy, causing it to match as few characters as possible.
      if (regex === "v") {
        pos = "verb";
      } else if (regex === "n") {
        pos = "noun";
      } else if (regex === "o") {
        pos = "object";
      } else if (regex === "a") {
        pos = "adjective";
      }
      return { word: word, pos: pos };
    } else {
      return { word: ele };
    }
  });
```

This is a more concise way of writing this block.